core/arch/arm/plat-ti/main.c

```diff
@@ -130,12 +93,16 @@ struct plat_nsec_ctx {
 
 void init_sec_mon(unsigned long nsec_entry)
 {
-	struct plat_nsec_ctx *plat_ctx = (struct plat_nsec_ctx *)nsec_entry;
+	struct plat_nsec_ctx *plat_ctx;
 	struct sm_nsec_ctx *nsec_ctx;
 
+	plat_ctx = phys_to_virt(nsec_entry, MEM_AREA_IO_SEC);
```

Comment:
Thinking more on the cache invalidation, I don't think it is strictly needed. As you said this area will be mapped un-cached as it is marked MEM_AREA_IO_SEC, but this is area is not IO area, it is just marked this way. A better fix would be to mark this area MEM_AREA_RAM_SEC, which it is, then the invalidation may be needed (this location is written by the platforms default TEE so I'm not sure about its cache state before the hand-off).